slave_stack/drivers/nuttx/SPI.cc

```diff
@@ -0,0 +1,34 @@
+#include "SPI.h"
+
+#include <fcntl.h>
+
+namespace kickcat
+{
+    void SPI::init()
+    {
+        fd_ = file_open(&filep_spi_, "/dev/spi0", O_RDWR);
+        printf("Open spi0 driver file fd: %d \n", fd_);
+
+        priv_spi_ = static_cast<spi_driver_s*>(filep_spi_.f_inode->i_private);
```

Comment:
Try to cast it to  spi_dev_s**

Reply:
it worked !

@trns1997 so we can get rid of 
```
    struct spi_driver_s
    {
        FAR struct spi_dev_s* spi; /* Contained SPI lower half driver */
    #ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
        mutex_t lock;              /* Mutual exclusion */
        int16_t crefs;             /* Number of open references */
        bool unlinked;             /* True, driver has been unlinked */
    #endif
    };
```